src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,301 @@
+            "Please set CSScorer object as scorer object of utility. See scoring argument of Utility."
+        )
+
+    result = ValuationResult.zeros(
+        algorithm="class_wise_shapley",
+        indices=indices,
+        data_names=u.data.data_names[indices],
+    )
+
+    _, y_train = u.data.get_training_data(indices)
+    unique_labels = np.unique(y_train)
+    pbar = tqdm(disable=not progress, position=0, total=100, unit="%")
+
+    while not done(result):
+
+        pbar.n = 100 * done.completion()
+        pbar.refresh()
+
+        for idx_label, label in enumerate([1, 0]):
+
+            u.scorer.label = label
+            result = __class_complement_conditional_sampler(
+                u,
+                label,
+                result,
+                active_indices=indices,
+                eps=eps,
+                num_samples=num_resample_complement_sets,
+            )
+
+    return result
+
+
+def class_wise_shapley(
+    u: Utility,
+    *,
+    progress: bool = False,
+    done: StoppingCriterion,
+    eps: float = 1e-4,
+    normalize_score: bool = True,
+    n_jobs: int = 4,
+    config: ParallelConfig = ParallelConfig(),
+) -> ValuationResult:
+    """
+    Computes the class-wise Shapley value using the formulation with permutations using map reduce.
+
+    :param u: Utility object with model, data, and scoring function. The scoring function has to be of type CSScorer.
+    :param progress: Whether to display progress bars for each job.
+    :param done: Criterion on when no new permutation shall be sampled.
+    :param normalize_score: Whether to normalize the score by the number of classes.
+    :param eps: The threshold when updating using the truncated monte carlo estimator.
+    :param n_jobs: Number of jobs to run in parallel.
+    :param config: Parallel configuration.
+    :return: ValuationResult object with the data values.
+    """
+
+    map_reduce_job: MapReduceJob[NDArray, ValuationResult] = MapReduceJob(
+        u.data.indices,
+        map_func=_class_wise_shapley_worker,
+        reduce_func=lambda results: reduce(operator.add, results),
+        map_kwargs=dict(u=u, done=done, progress=progress, eps=eps),
+        n_jobs=n_jobs,
+        config=config,
+    )
+    result = map_reduce_job()
+    y_train = u.data.y_train
+    unique_labels = np.unique(np.concatenate((y_train, u.data.y_test)))
+
+    if normalize_score:
+
+        for idx_label, label in enumerate(unique_labels):
+
+            u.scorer.label = label
+            active_elements = y_train == label
+            label_set = np.where(active_elements)[0]
+
+            u.model.fit(u.data.x_train, u.data.y_train)
+            in_cls_acc, _ = _estimate_in_out_cls_accuracy(
+                u.model, u.data.x_test, u.data.y_test, label
+            )
+
+            sigma = np.sum(result.values[label_set])
+            result.values[label_set] *= in_cls_acc / sigma
+
+    return result
+
+
+def __class_complement_conditional_sampler(
```

Comment:
Double underscores have a special meaning in python. Use just one, or in this case probably none. Also, this function is called `...sampler` but it does some additional computation, so it's a misnomer.

Finally, passing a fixed number of samples is not very robust nor flexible. That's why we have `StoppingCriteria`. You can also use them for an inner loop like this, possibly a separate criterion.